CMakeLists.txt

```diff
@@ -29,9 +29,15 @@ include("${PROJECT_SOURCE_DIR}/cmake/functions.cmake")
 google_initialize_cartographer_project()
 google_enable_testing()
 
+find_package(Abseil REQUIRED)
 find_package(Boost REQUIRED COMPONENTS iostreams)
 find_package(Ceres REQUIRED COMPONENTS SuiteSparse)
-find_package(Eigen3 REQUIRED)
+
+find_package(Eigen3)
+if(NOT EIGEN3_FOUND)
+  set(EIGEN3_INCLUDE_DIRS ${/usr/include/eigen3})
```

Comment:
is this unrelated to absl?

Reply:
When I install cmake 3.x in trusty/jessie, Cartographer complains that it cannot find FindEigen3.cmake, although Ceres was compiled just prior to that. I do not know whether it is a good solution or not. @gaschler PTAL, because CMake scares and confuses me.

Reply:
This is unrelated to abseil and should be a separate PR where only the cmake version is bumped.
(It's hard to say why a newer cmake makes the package search stricter.)

So, no change here, please. (Setting an include path manually is not portable and wouldn't be part of a cmake style guide.)
A proper solution would be to add a FindEigen3.cmake to cmake/modules similar to (but much shorter than) what ceres does: https://github.com/ceres-solver/ceres-solver/blob/master/cmake/FindEigen.cmake

Reply:
Also, when bumping cmake, it would be good to bump `cmake_minimum_required` in this file and update the install instructions in the documentation.